### PR TITLE
Asset and doc compiling in CI

### PR DESCRIPTION
**Pull request by JoshAshby**
This is an alternative to #9 and a holdover until I get custom docker images with the assets in them. It should compile the API docs and assets in CI and then push a tar of them into a digitalocean space, where I can have the docker build download them during a deploy.

### CODE REVIEW

**Comment by JoshAshby** [comment]
Goes with https://github.com/transientBug/ansible/pull/1 for actual deploys

File: .travis.yml [hunk 1]
```diff
@@ -42,7 +42,13 @@ jobs:
...
+
     - stage: stylistic
       script: bundle exec pronto run -f github_status -c origin/master
+
     - stage: docs
       script: bundle exec yard
+
+    - stage: compile
+      script: .travis/assets
+      #if: branch = master
```
Comment (robacarp) [inline .travis.yml]: if

File: .travis/assets [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Building API docs"
+bundle exec rake docs:generate
+
+echo "Building new assets"
+env RAILS_ENV=production bundle exec rake assets:precompile
+
+export DO_SPACES_ENDPOINT=https://nyc3.digitaloceanspaces.com
+export CURRENT_DATE=$(date -u +%FT%H%M%S)
+export PACKAGE=package-${CURRENT_DATE}.tar.gz
```
Comment (robacarp) [inline .travis/assets]: you need quotes everywhere in this file

File: .travis/assets [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Building package $PACKAGE"
+tar -zcvf $PACKAGE public doc
```
Comment (robacarp) [inline .travis/assets]: "

File: .travis/assets [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+./.travis/common
+
+echo "Building API docs"
+bundle exec rake docs:generate
+
+echo "Building new assets"
+env RAILS_ENV=production bundle exec rake assets:precompile
+
+export DO_SPACES_ENDPOINT=https://nyc3.digitaloceanspaces.com
+export CURRENT_DATE=$(date -u +%FT%H%M%S)
```
Comment (robacarp) [inline .travis/assets]: "

File: .travis/assets [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+./.travis/common
+
+echo "Building API docs"
+bundle exec rake docs:generate
+
+echo "Building new assets"
+env RAILS_ENV=production bundle exec rake assets:precompile
+
+export DO_SPACES_ENDPOINT=https://nyc3.digitaloceanspaces.com
```
Comment (robacarp) [inline .travis/assets]: "

File: .travis/assets [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+export CURRENT_DATE=$(date -u +%FT%H%M%S)
+export PACKAGE=package-${CURRENT_DATE}.tar.gz
+
+echo "Building package $PACKAGE"
+tar -zcvf $PACKAGE public doc
+
+echo "Installing awscli"
+pip install --user awscli # Install awscli w/o sudo
+export PATH=$PATH:$HOME/.local/bin # Ensure we can call awscli
```
Comment (robacarp) [inline .travis/assets]: mosdef "

File: .travis/assets [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Uploading $PACKAGE to DO space"
+aws s3 cp $PACKAGE s3://transientbug-assets/ --endpoint $DO_SPACES_ENDPOINT
+aws s3 cp $PACKAGE s3://transientbug-assets/current_package.tar.gz --endpoint $DO_SPACES_ENDPOINT
```
Comment (robacarp) [inline .travis/assets]: "

File: .travis/assets [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Building package $PACKAGE"
+tar -zcvf $PACKAGE public doc
+
+echo "Installing awscli"
+pip install --user awscli # Install awscli w/o sudo
+export PATH=$PATH:$HOME/.local/bin # Ensure we can call awscli
+
+echo "Uploading $PACKAGE to DO space"
+aws s3 cp $PACKAGE s3://transientbug-assets/ --endpoint $DO_SPACES_ENDPOINT
```
Comment (robacarp) [inline .travis/assets]: "

File: .travis/common [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+echo "Creating test database"
+psql -c 'create database travis_ci_test;' -U postgres
+
+echo "Starting ES"
+./elasticsearch-${ES_VERSION}/bin/elasticsearch &
+
+echo "Yarn install"
+yarn install
+
+echo "Moving CI Master key into place"
```
Comment (robacarp) [inline .travis/common]: " brah, "

File: .travis/assets (hunk 2 above)
Comment (JoshAshby) [reply .travis/assets]: so I don't, this works fine

**Pull request merged**